contracts/modules/Wallet/VestingEscrowWallet.sol

```diff
@@ -0,0 +1,525 @@
+        emit DepositTokens(_numberOfTokens, now);
+    }
+
+    /**
+     * @notice Sends unassigned tokens to treasury
+     */
+    function sendToTreasury() external withPerm(ADMIN) {
+        uint256 amount = unassignedTokens;
+        unassignedTokens = 0;
+        token.safeTransfer(treasury, amount);
+        /*solium-disable-next-line security/no-block-members*/
+        emit SendToTreasury(amount, now);
+    }
+
+    /**
+     * @notice Sends available tokens to beneficiary
+     * @param _beneficiary beneficiary's address
+     */
+    function sendAvailableTokens(address _beneficiary) public withPerm(ADMIN) {
+        _sendTokens(_beneficiary);
+    }
+
+    /**
+     * @notice Used to withdraw available tokens by beneficiary
+     */
+    function withdrawAvailableTokens() external {
+        _sendTokens(msg.sender);
+    }
+
+    /**
+     * @notice Add template
+     * @param _numberOfTokens number of tokens
+     * @param _duration vesting duration
+     * @param _frequency vesting frequency
+     */
+    function addTemplate(uint256 _numberOfTokens, uint256 _duration, uint256 _frequency) external withPerm(ADMIN) {
+        _validateTemplate(_numberOfTokens, _duration, _frequency);
+        Template memory template;
+        template.numberOfTokens = _numberOfTokens;
+        template.duration = _duration;
+        template.frequency = _frequency;
+        templates.push(template);
+        /*solium-disable-next-line security/no-block-members*/
+        emit AddTemplate(_numberOfTokens, _duration, _frequency, now);
+    }
+
+    /**
+     * @notice Removes template
+     * @param _index index of the template
+     */
+    function removeTemplate(uint256 _index) external withPerm(ADMIN) {
+        require(_index < templates.length, "Template not found");
+        templates[_index] = templates[templates.length - 1];
+        templates.length--;
+        /*solium-disable-next-line security/no-block-members*/
+        emit RemoveTemplate(_index, now);
+    }
+
+    /**
+     * @notice Returns count of templates
+     * @return count of templates
+     */
+    function getTemplateCount() external view withPerm(ADMIN) returns(uint256) {
+        return templates.length;
+    }
+
+    /**
+     * @notice Adds vesting schedules for each of beneficiary
+     * @param _beneficiary beneficiary's addresses
+     * @param _numberOfTokens number of tokens
+     * @param _duration vesting duration
+     * @param _frequency vesting frequency
+     * @param _startTime vesting start time
+     */
+    function addSchedule(
+        address _beneficiary,
+        uint256 _numberOfTokens,
+        uint256 _duration,
+        uint256 _frequency,
+        uint256 _startTime
+    )
+        public
+        withPerm(ADMIN)
+    {
+        _validateSchedule(_beneficiary, _numberOfTokens, _duration, _frequency, _startTime);
+        require(_numberOfTokens <= unassignedTokens, "Not enough tokens");
+
```

Comment:
we can save one call, using this
      
        if (_numberOfTokens <= unassignedTokens) {
                  _depositTokens() -- internal version of the depositTokens
          }
This will avoid explicit calling of `depositTokens()`